contracts/PartyBidFactory.sol

```diff
@@ -0,0 +1,122 @@
+ * A poorly implemented MarketWrapper contract could permanently lose access to the NFT.
+ * When deploying a PartyBid, exercise extreme caution.
+ * Only use MarketWrapper contracts that have been audited and tested.
+ */
+contract PartyBidFactory {
+    //======== Events ========
+
+    event PartyBidDeployed(
+        address partyBidProxy,
+        address creator,
+        address nftContract,
+        uint256 tokenId,
+        address marketWrapper,
+        uint256 auctionId,
+        address partyHost,
+        uint256 partyHostBasisPoints,
+        string name,
+        string symbol
+    );
+
+    //======== Immutable storage =========
+
+    address public immutable logic;
+    address public immutable partyDAOMultisig;
+    address public immutable tokenVaultFactory;
+    address public immutable weth;
+
+    //======== Mutable storage =========
+
+    // PartyBid proxy => block number deployed at
+    mapping(address => uint256) public deployedAt;
+
+    //======== Constructor =========
+
+    constructor(
+        address _partyDAOMultisig,
+        address _tokenVaultFactory,
+        address _weth,
+        address _logicMarketWrapper,
+        address _logicNftContract,
+        uint256 _logicTokenId,
+        uint256 _logicAuctionId
+    ) {
+        partyDAOMultisig = _partyDAOMultisig;
+        tokenVaultFactory = _tokenVaultFactory;
+        weth = _weth;
+        // deploy logic contract
+        PartyBid _logicContract = new PartyBid(_partyDAOMultisig, _tokenVaultFactory, _weth);
+        // initialize logic contract
+        _logicContract.initialize(
+            _logicMarketWrapper,
+            _logicNftContract,
+            _logicTokenId,
+            _logicAuctionId,
+            address(0),
+            0,
+            "PartyBid",
+            "BID"
+        );
+        // store logic contract address
+        logic = address(_logicContract);
+    }
+
+    //======== Deploy function =========
+
+    function startParty(
+        address _marketWrapper,
+        address _nftContract,
+        uint256 _tokenId,
+        uint256 _auctionId,
+        address _partyHost,
+        uint256 _partyHostBasisPoints,
+        string memory _name,
+        string memory _symbol
+    ) external returns (address partyBidProxy) {
+        bytes memory _initializationCalldata =
+            abi.encodeWithSignature(
+                "initialize(address,address,uint256,uint256,address,uint256,string,string)",
+                _marketWrapper,
+                _nftContract,
+                _tokenId,
+                _auctionId,
+                _partyHost,
+                _partyHostBasisPoints,
+                _name,
+                _symbol
+            );
+
+        partyBidProxy = address(
+            new InitializedProxy(
+                logic,
+                _initializationCalldata
+            )
+        );
+
+        deployedAt[partyBidProxy] = block.number;
```

Comment:
What's the point of this?

Reply:
it allows folks on-chain or off-chain to query for a PartyBid address to see if it was deployed by the canonical Factory. the choice to record block number is somewhat arbitrary